### PR TITLE
Update the navigation-timing IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Review by igrigorik — COMMENTED** [review]
Sanity check: we no longer have to define PerformanceEntry and PerformanceResourceTiming here?

File: navigation-timing/idlharness.window.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// META: script=/resources/WebIDLParser.js
+// META: script=/resources/idlharness.js
+
+'use strict';
+
+// http://www.w3.org/TR/navigation-timing/
+
+idl_test(
+  ['navigation-timing'],
+  ['resource-timing', 'performance-timeline', 'hr-time', 'dom'],
```
Comment (lukebjerring) [inline navigation-timing/idlharness.window.js]: @igrigorik - correct, `PerformanceEntry` and `PerformanceResourceTiming` are defined in `hr-time` + `performance-timeline` specs. Previously, they were duplicated, under the `<pre id='untested_idl' style='display:none'>` tag (untested).

This is now handled by `add_dependency_idls` which only calls `add_untested_idls` for any IDL which it has detected to be a dependency of the main IDL (in this case, `navigation-timing`). This line here is the `dependency specs` param of the `idl_test` helper, meaning that each spec in this second array is passed to `add_dependency_idls`.

Comment (igrigorik) [reply navigation-timing/idlharness.window.js]: @lukebjerring cool, thanks!

**Review by igrigorik — APPROVED** [review]
👍

**Pull request merged**